xDripG5/BluetoothManager.swift

```diff
@@ -149,7 +149,7 @@ class BluetoothManager: NSObject {
             self.scanForPeripheral(after: TimeInterval(60 * 3))
         } else {
             DispatchQueue.global(qos: DispatchQoS.QoSClass.utility).async {
-                Thread.sleep(forTimeInterval: 2)
+                Thread.sleep(forTimeInterval: 60 * 3)
```

Comment:
No, it’s not OK to Thread.sleep for 180 seconds.

Reply:
Why don't you just use asyncAfter(deadline: .now() + 60*3)?

Reply:
> Why don't you just use asyncAfter(deadline: .now() + 60*3)?

It doesn't matter what form of wait is used, really. CoreBluetooth is asking the system to continue to let the app run in the background for a few seconds at a time, and when that times out, any scheduled execution won't be called.

A blocking call like Thread.sleep ensures forces the system to kill the app if it's unhappy with the length, which is a more apparent failure mode than a timer-based async approach.

Reply:
OK, thanks for the explanation of the 2 second wait. It wasn't apparent to me that the wait is to give the receiver a chance to connect. What prompted this for me was that the example app very briefly displays a glucose value, followed by "Error". This is because the delegate method `transmitter(didRead: glucose)` is called followed by `transmitter(didError: error)` when the phone tries to reconnect but is unsuccessful.

Would you consider a PR that suppresses the `didError` call for the unsuccessful reconnect?